osquery/tables/system/windows/smbios_tables.cpp

```diff
@@ -12,11 +12,96 @@
 
 #include <osquery/core/tables.h>
 #include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/windows/strings.h>
 
 #include "osquery/core/windows/wmi.h"
 
 namespace osquery {
 namespace tables {
+namespace {
+
+const std::vector<std::string> kFormFactors = {
+    "Unknown",     "Other", "SIP",  "DIP",  "ZIP",   "SOJ",
+    "Proprietary", "SIMM",  "DIMM", "TSOP", "PGA",   "RIMM",
+    "SODIMM",      "SRIMM", "SMD",  "SSMP", "QFP",   "TQFP",
+    "SOIC",        "LCC",   "PLCC", "BGA",  "FPBGA", "LGA",
+};
+
+std::string getFormFactor(long id) {
+  if (id < kFormFactors.size()) {
+    return kFormFactors[id];
+  }
+  return std::to_string(id);
+}
+
+const std::vector<std::string> kMemoryTypes = {
+    "Unknown",    "Other",  "DRAM",         "Synchronous DRAM",
+    "CACHE DRAM", "EDO",    "EDRAM",        "VRAM",
+    "SRAM",       "RAM",    "ROM",          "Flash",
+    "EEPROM",     "FEPROM", "EPROM",        "CDRAM",
+    "3DRAM",      "SDRAM",  "SGRAM",        "RDRAM",
+    "DDR",        "DDR2",   "DDR2 FB-DIMM", "23",
+    "DDR3",       "FBD2",   "DDR4"};
+
+std::string getMemoryType(int id) {
+  if (id < kMemoryTypes.size()) {
+    return kMemoryTypes[id];
+  }
+  return std::to_string(id);
+}
+
+std::string getMemoryTypeDetails(int id) {
```

Comment:
Why lookup tables above, and a case statement here? Either seems fine, but both feels weird?

Reply:
Oh. I see now. And if I dig a bit, Microsoft documents this as coming from SMBIOS.

I see that we have function in `osquery/tables/system/posix/smbios_utils.cpp` for the data that is from SMBIOS, does it make sense to unify the helpers?

Reply:
I don't see any real benefit, and unifying seems like a non-trivial task as some code may be platform specific still.

Reply:
Okay. I don't love the duplication, but I concede that some duplication is better than early optimization.  Mostly I think about the user side -- there's value in ensuring the reported values are consistent across platforms. (If they're all defined by SMBIOS)